tools/tertiary-analysis/data-scxa/retrieve-scxa.xml

```diff
@@ -19,8 +19,8 @@ cut -f2 '${accession}'.expression_tpm.mtx_cols > ${barcode_tsv};
 
 wget -O ${matrix_mtx} 'ftp://ftp.ebi.ac.uk/pub/databases/microarray/data/atlas/sc_experiments/${accession}/${accession}.aggregated_filtered_counts.mtx';
 wget -qO - 'ftp://ftp.ebi.ac.uk/pub/databases/microarray/data/atlas/sc_experiments/${accession}/${accession}.aggregated_filtered_counts.mtx_cols' | cut -f2 > ${barcode_tsv};
-wget -qO - 'ftp://ftp.ebi.ac.uk/pub/databases/microarray/data/atlas/sc_experiments/${accession}/${accession}.aggregated_filtered_counts.mtx_rows'
- | awk '{OFS="\t"; print \$2,\$2}' > ${genes_tsv};
+wget -qO - 'ftp://ftp.ebi.ac.uk/pub/databases/microarray/data/atlas/sc_experiments/${accession}/${accession}.aggregated_filtered_counts.decorated.mtx_rows' |
+  awk -F'\t' '{ if (length($2) == 0) { print $1"\t"$1 } else { print $0 } }' > ${genes_tsv};
```

Comment:
Hmm, didn't know that this '$' in '$0', '$1', '$2' doesn't need escaping.

Reply:
Yes, interesting now that you mention it... I guess that if cheetah doesn't find a value to replace with (a variable called 0, 1, 2) it doesn't do it? I did run it on conda and on containers and seems to be working fine (awk would have given an error and the files would be empty if the replacement happened by empty values). Should we still escape them?

Reply:
It looks good to me the way it is. The documentation for parsing of the command section of Galaxy tool xml is somewhat elusive, so I was learning by example.